src/worker.ts

```diff
@@ -12,30 +12,27 @@ function markets(orderbooksString: string, base: string, quote: string, hops: nu
 }
 
 function estimatedBuyAmount(
-  orderbooksString: string,
-  base: string,
-  quote: string,
-  hops: number,
-  quoteAmount: string,
-  priceRoundingBuffer: number,
+  encodedOrders: Uint8Array,
+  base: number,
+  quote: number,
+  sellAmountInQuote: number,
```

Comment:
This is of type number but in the test it gets compared to a string (cf. changed lines that is adding quotes). Why is that?

Reply:
I think it should probably be a string for consistency with the other BN values - and some special care should be taken to make sure it isn't formatted in scientific notations (easiest way off the top of my head would be `BigInt(value).toString()`).

Reply:
ubernit: I think this should probably stay a string. The reasoning being the API operates on big integers which cannot be correctly expressed within an number. Ultimately, it will get converted to a number to use for the pricegraph, but I think that should happen inside the worker, as it is an implementation detail about how the estimate gets calculated.

Reply:
Why does the test need a string now? I am not sure. Both in the old and new version we are storing a number according to typescript but then somehow it turns into a string only in the new version.
I'll do what Nick said and make all the numbers always strings.

Reply:
It looks like the `sellAmountInQuote` used to be
https://github.com/gnosis/dex-price-estimator/blob/8888b06677dd544b0d6906980c113c12d79e87f3/src/worker.ts#L19

And would get parsed into a number on return:
https://github.com/gnosis/dex-price-estimator/blob/8888b06677dd544b0d6906980c113c12d79e87f3/src/worker.ts#L38